src/typescript/frontend/src/components/pages/emojicoin/components/main-info/MainInfo.tsx

```diff
@@ -52,6 +82,61 @@ const MainInfo = ({ data }: MainInfoProps) => {
 
   const [copied, setCopied] = useState(false);
 
+  const borderStyle = "border-solid border-[1px] border-dark-gray rounded-[3px] p-[1em]";
+
+  const [marketProperties, setMarketProperties] = useState<Map<string, string>>();
+
+  const { aptos } = useAptos();
+
+  useEffect(() => {
+    MarketProperties.submit({
```

Comment:
You can change this function name to `view` btw, meant to change it when generating the APIs

Reply:
Addressed in 6ebc11c1a46c8e30cae4600360b55137e38a6a26.